src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java

```diff
@@ -469,6 +476,56 @@ void hanldeTopicCreateOperation(final O topicOp, final Map<String, KafkaFuture<V
     topicOp.finishRequest();
   }
 
+  void readTopics(final List<O> topicOps) {
+    val nodeAddr = topicOps.get(0).nodeAddr();
+    try {
+      concurrencyThrottle.acquire();
+      for (var i = 0; i < topicOps.size(); i++) {
+        val topicOp = (PathOperation) topicOps.get(i);
+        val topicItem = topicOp.item();
+        val topicName = topicItem.name().replaceAll("/", "");
+        try {
+          val consumerConfig = createConsumerConfig(nodeAddr);
+          consumerConfig.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, Integer.MAX_VALUE);
+          topicOp.startRequest();
+          try (val kafkaConsumer = new KafkaConsumer<String, byte[]>(consumerConfig)) {
+            kafkaConsumer.subscribe(Arrays.asList(topicName));
+            topicOp.finishRequest();
+            topicOp.startResponse();
+            var sizeOfReadData = 0;
+            var records = kafkaConsumer.poll(recordOpTimeout);
+            for (var record : records) {
```

Comment:
Why can't we delete this for-loop?

Reply:
Because I need to count bytes from the first poll too. The while circle is made for polling records until the current topic become empty.

Reply:
Done